### ISSUE TITLE
[Bug]: OpenMPI and CUDA buffers

### ISSUE THREAD

**Issue by JuanPedroGHM** [helmholtz-analytics/heat#1953]
### What happened?

On paper, OpenMPI 5.0 and up supports communicating CUDA buffers, but there are some issues inside heat when using CUDA buffers.

The goal of the issues is to document where errors happen and to track the sub-issues until we have more stable GPU support.

Steps taken:
- Disabled CUDA buffers for OpenMPI in #1926 

Currently known issues:
 - QR unit tests: FTPx86 Mi250 and Horeka 
 - Benchmarks: FTPx86 Mi250 (needs more investigation)
 - Cannot update to most recent MPI Action: #1351 
- Other? ...

### Code snippet triggering the error

```shell

```

### Error message or erroneous outcome

```shell

```

### Version

main (development branch)

### Python version

3.12

### PyTorch version

2.7

### MPI version

```shell
OpenMPI
```

**Comment by JuanPedroGHM** [comment]
From the Support Pytorch 2.9.1 PR #2001 :
 

> Tests on Horeka (Single Node, Accelerated, H100, Python 3.12.1)
> 
>     OpenMPI 5.0
>         Works with both CPU and GPU.
>     IntelMPI 2021.11
>         Works for 1 and 2 ranks, crashes during test_svdtools.py for 3 and 4.
> 
> ===================================================================================
> =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
> =   RANK 1 PID 4089957 RUNNING AT hkn0808
> =   KILLED BY SIGNAL: 9 (Killed)
> ===================================================================================
> 
> ===================================================================================
> =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
> =   RANK 2 PID 4089958 RUNNING AT hkn0808
> =   KILLED BY SIGNAL: 9 (Killed)
> ===================================================================================
> 
>     MVAPICH 4.0
>         CPU: Works on 1 and 2 ranks, get's stuck on test_svd.py with 4 ranks.
>         GPU: Works on 1 and 2 ranks, crashes on test_save_csv.py and fails test_svd.py and test_svdtools.py
> 
> Current thread 0x000014d03055ab80 (most recent call first):
>  File "/hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/core/io.py", line 1123 in save_csv
>  File "/hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/core/io.py", line 1250 in save
>  File "/hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/core/io.py", line 1260 in <lambda>
>  File "/hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/core/tests/test_io.py", line 246 in test_save_csv
>  File "/usr/lib64/python3.12/unittest/case.py", line 589 in _callTestMethod
> 
> Warnings (Everywhere)
> 
> /hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/sparse/tests/test_arithmetics_csr.py:605: UserWarning: To copy construct
>  from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than t
> orch.tensor(sourceTensor).
>     == torch.tensor(indices_C, device=self.device.torch_device)
> 
>   /hkfs/home/project/hk-project-test-aihero2/io3047/code/heat/heat/core/statistics.py:305: UserWarning: Using a non-tuple sequence for mult
> idimensional indexing is deprecated and will be changed in pytorch 2.9; use x[tuple(seq)] instead of x[seq]. In pytorch 2.9 this will be in
> terpreted as tensor index, x[torch.tensor(seq)], which will result either in an error or a different result (Triggered internally at /pytor
> ch/torch/csrc/autograd/python_variable_indexing.cpp:345.)
>     wgt[wgt_slice] = weights.larray
>
